### PR TITLE
chore: extract email metadata parsing and preamble builders from dispatcher (#465)

### LINKED ISSUE #465: Refactor dispatcher: extract email metadata parsing and preamble-building into testable units
## Problem

`handleInbound` in `src/dispatch/dispatcher.ts` has accumulated several concerns that are tangled together and untestable in isolation:

**1. Repeated raw metadata casts**
`payload.metadata as Record<string, unknown> | undefined` appears six times across the method (hold paths at lines ~341, ~457, ~679; thread-participants at ~725; CC preamble at ~767; and `mergeTaskMetadata` at ~881). Each cast is a silent boundary where a mistyped field produces `undefined` at runtime with no type error at compile time.

**2. Inline `nylasMessageId` extraction and sanitization**
The CC preamble path (lines ~797-825) casts, sanitizes, and validates `nylasMessageId` inline with detailed warn logging. This logic is tightly coupled to the dispatcher method — any change to the extraction pattern requires reading the full method to understand context. There is no shared helper enforcing safe extraction.

**3. Preamble and context-block building is untestable**
Both the CC role preamble (lines ~763-843) and the thread-participants block (lines ~721-761) are embedded inline. The only way to exercise them today is to run the full dispatcher with a mocked bus and faked payload — expensive, brittle, and tests far more than the formatting logic. `buildContextPreamble` was already extracted to `context-memo.ts`, establishing a precedent for this pattern.

## Proposed shape

The refactor should not change any observable behavior — it is a structural improvement only.

- **Typed `EmailMetadata` interface** covering the known email-specific fields (`nylasMessageId`, `curiaRole`, `primaryRecipientEmails`, `participants`, `subject`, etc.). Extracted once per incoming email via a small parse helper; subsequent code reads typed fields, not repeated casts.
- **`sanitizeNylasMessageId(raw: unknown, logger: Logger, context: object): string | undefined`** — single function for the strip-trim-truncate-warn pattern, replacing the inline implementation.
- **`buildCcPreamble(meta: EmailMetadata, accountId: string | undefined): string`** — pure function producing the `[OWNER CC …]` block with Message ID and Account.
- **`buildThreadParticipantsBlock(meta: EmailMetadata, selfEmail: string | undefined): string | null`** — pure function producing the `[Thread participants …]` header.
- Unit tests for all helpers covering: ID present, ID absent, ID sanitizes to empty, participant list edge cases, recipient list formatting including truncation.

## Acceptance criteria

- [ ] `payload.metadata` is cast to a typed `EmailMetadata` structure **once** per email message, at the top of the email-handling section; subsequent code reads typed fields, not repeated casts
- [ ] `nylasMessageId` extraction and sanitization lives in one shared function with its own unit tests
- [ ] The CC role preamble block is a pure function (or private method taking only its inputs) that can be unit-tested without instantiating a `Dispatcher`
- [ ] The thread-participants block is similarly extracted into a testable pure function
- [ ] Unit tests cover: correct output when Message ID is present, absent, or sanitizes to empty; correct recipient list formatting including truncation; participant list with self-email replacement, max-participant cap, and empty inputs
- [ ] No change to observable dispatcher behavior — existing dispatcher tests continue to pass unchanged

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- Extracts `parseEmailMetadata`, `sanitizeNylasMessageId`, `buildCcPreamble`, and `buildThreadParticipantsBlock` from `handleInbound` into a new `src/dispatch/email-metadata.ts` module
- The two separate `if (payload.channelId === 'email')` blocks in `handleInbound` are collapsed into one, parsing metadata once via `parseEmailMetadata` and delegating to the pure helpers
- `sanitizeNylasMessageId` returns a discriminated result (`{ ok: true; value } | { ok: false; reason }`) so the dispatcher can log specific failure reasons without a logger dependency in the helper module
- 34 new unit tests following the `context-memo.test.ts` pattern
- No change to observable dispatcher behaviour — all existing dispatch tests continue to pass

**Security hardening picked up during review:**
- `sanitizeEmailField` now guards `null`/`undefined` inputs (previously `String(null)` would have produced the literal string `"null"` as a recipient address in the CC preamble)
- The participants loop validates each element is a non-null object before property access, guarding against null array elements from Nylas schema drift

Closes #465

## Test plan

- [ ] All 211 dispatch unit tests pass (`tests/unit/dispatch/`)
- [ ] New tests cover: `parseEmailMetadata` (typed extraction, safe defaults, non-string/non-array field handling), `sanitizeNylasMessageId` (clean ID, injection stripping, truncation, all failure reasons), `buildCcPreamble` (all recipient edge cases, Message ID present/absent, sanitization), `buildThreadParticipantsBlock` (participants formatting, null elements, self-email substitution, MAX_PARTICIPANTS cap)
- [ ] CI green

## Summary by CodeRabbit

* **Refactor**
  * Extracted email metadata processing logic into dedicated helper functions to improve code maintainability and reduce duplication across email handling workflows
  * Reorganized dispatcher implementation for clearer separation of concerns

* **Tests**
  * Added 34 comprehensive unit tests verifying email metadata extraction, validation, sanitization, and preamble generation

[Review Change Stack](https://app.coderabbit.ai/change-stack/josephfung/curia/pull/600)

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai review

File: src/dispatch/email-metadata.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+  // Count items dropped by the empty-string filter or by the 10-item cap.
+  const omittedCount = Math.max(0, meta.primaryRecipientEmails.length - sanitizedRecipients.length);
+
+  const recipientListBase = sanitizedRecipients.length > 0
+    ? sanitizedRecipients.join(', ')
+    : 'unknown recipients';
+  const recipientList = omittedCount > 0
+    ? `${recipientListBase}, +${omittedCount} more`
+    : recipientListBase;
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/dispatch/email-metadata.ts]: Fixed in 0d27dc3. Collapsed the two-step build into a single conditional so the "+N more" suffix only appears when `sanitizedRecipients.length > 0`. Added a regression test covering the all-sanitized-to-empty case.

**Comment by josephfung** [comment]
Addressing the two nitpicks from the review:

**CHANGELOG entry** — condensed in 0d27dc3. Down to ~15 words after the em-dash.

**`MAX_PARTICIPANTS` slice before validation** — skipping. This is the same order as the original dispatcher code (`rawParticipants.slice(0, 15)` then the null guard in the loop), carried over unchanged. Changing it would be a behaviour difference with no functional benefit in practice (Nylas participant lists are typically small and well-formed).

**Comment by josephfung** [comment]
@coderabbitai approve
